storage/spanstore/automate_dropping_writer.go

```diff
@@ -0,0 +1,33 @@
+package spanstore
+
+import (
+	"math"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// AutomateDroppingWriter is a span Writer that tries to tries to drop spans with
+// percentage set in config file
+type AutomateDroppingWriter struct {
+	spanWriter Writer
+	threshold  uint64
+}
+
+// NewAutomateDroppingWriter creates a AutomateDroppingWriter
+func NewAutomateDroppingWriter(spanWriter Writer, percentage float64) *AutomateDroppingWriter {
+	// span with TraceID.Low below this threshold won't be written
+	threshold := uint64(percentage * float64(math.MaxUint64))
+	return &AutomateDroppingWriter{
+		spanWriter: spanWriter,
+		threshold:  threshold,
+	}
+}
+
+// WriteSpan calls WriteSpan on wrapped span writer.
+func (c *AutomateDroppingWriter) WriteSpan(span *model.Span) error {
+	// TraceID consists of two random generated uint64, we use one of them to decide if span will be written
+	if c.threshold != 0 && span.TraceID.Low < c.threshold {
```

Comment:
No, let's not do that. This type of logic is too easy to write and thus is being used elsewhere, in particular in the clients that try to minimize computations.

A better approach is to compute a hash of the trace ID with some fixed salt and compare the hash against the ratio. Btw that kind of logic can be extracted into a util type.

Reply:
We are not doing encryption/decryption. Is salt really necessary here?

Reply:
yes, because many components in the collection chain might be doing downsampling, theoretically using the same algorithm. Using a salt provides a different distribution of the hash values.